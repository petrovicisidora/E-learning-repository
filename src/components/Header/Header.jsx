//Header

import React from "react";

import logo from "../../assets/logo.png";
import { StyledHeaderContainer, StyledHeaderLogo } from "./Header.style.js";

function Header() {
  return (
    <StyledHeaderContainer role="header">
      <div style={{ marginLeft: "auto" }}>
        <StyledHeaderLogo src={logo} alt="Logo" />
      </div>
    </StyledHeaderContainer>
  );
}

export default Header;
