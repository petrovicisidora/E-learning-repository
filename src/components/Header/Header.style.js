import styled from "styled-components";

export const StyledHeaderContainer = styled.header`
  width: 100%;
  height: 60px;
  background-color: #135cc3eb;
  display: flex;
`;

export const StyledHeaderLogo = styled.img`
  height: 60px;
  width: auto;
`;
