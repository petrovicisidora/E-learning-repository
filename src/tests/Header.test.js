import React from "react";
import { render, screen } from "@testing-library/react";

import Header from "../components/Header/Header";

test("renders header's logo correctly", () => {
  render(<Header />);
  const headerLogo = screen.getByAltText("Logo");

  expect(headerLogo).toBeInTheDocument();
});

test("renders header's blue background correctly", () => {
  render(<Header />);
  const headerContainer = screen.getByRole("header");

  expect(headerContainer).toHaveStyle("background-color: #135cc3eb");
});

test("matches snapshot", () => {
  const { container } = render(<Header />);
  expect(container).toMatchSnapshot();
});
