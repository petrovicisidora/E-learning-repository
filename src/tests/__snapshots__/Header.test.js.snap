// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <header
    class="sc-beySPh erKIBm"
    role="header"
  >
    <div
      style="margin-left: auto;"
    >
      <img
        alt="Logo"
        class="sc-guDLey bRskqZ"
        src="logo.png"
      />
    </div>
  </header>
</div>
`;
